envoy-control-core/src/test/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/NodeMetadataValidatorTest.kt

```diff
@@ -9,17 +9,67 @@ import org.junit.jupiter.api.Test
 import org.junit.jupiter.params.ParameterizedTest
 import org.junit.jupiter.params.provider.CsvSource
 import pl.allegro.tech.servicemesh.envoycontrol.snapshot.EnabledCommunicationModes
+import pl.allegro.tech.servicemesh.envoycontrol.snapshot.IncomingPermissionsProperties
 import pl.allegro.tech.servicemesh.envoycontrol.snapshot.OutgoingPermissionsProperties
 import pl.allegro.tech.servicemesh.envoycontrol.snapshot.SnapshotProperties
 
 class NodeMetadataValidatorTest {
 
     val validator = NodeMetadataValidator(SnapshotProperties().apply {
-        outgoingPermissions = createOutgoingPermissions(enabled = true, servicesAllowedToUseWildcard = mutableSetOf("vis-1", "vis-2"))
+        outgoingPermissions = createOutgoingPermissions(
+                enabled = true,
+                servicesAllowedToUseWildcard = mutableSetOf("vis-1", "vis-2")
+        )
+        incomingPermissions = createIncomingPermissions(
+                enabled = true,
+                servicesAllowedToUseWildcard = mutableSetOf("vis-1", "vis-2")
+        )
     })
 
     @Test
-    fun `should fail if service has no privilege to use wildcard`() {
+    fun `should fail if service has no privilege to use incoming wildcard`() {
+        // given
+        val node = node(
+                serviceName = "regular-1",
+                incomingSettings = true,
+                clients = listOf("*")
+        )
+        val request = DiscoveryRequest.newBuilder().setNode(node).build()
+
+        // expects
+        assertThatExceptionOfType(WildcardPrincipalValidationException::class.java)
+                .isThrownBy { validator.onStreamRequest(streamId = 123, request = request) }
+                .satisfies {
+                    assertThat(it.status.description).isEqualTo(
+                            "Blocked service regular-1 from allowing everyone in incoming permissions. Only defined services can use that."
+                    )
+                    assertThat(it.status.code).isEqualTo(Status.Code.INVALID_ARGUMENT)
+                }
```

Comment:
How about splitting `expects` section to have more of a given/when/then structure like everywhere else. Also `satisfies` produces not very readable error messages.


```suggestion
        // when
        val exception = catchThrowable { validator.onStreamRequest(streamId = 123, request = request) }
        
        // then
        assertThat(exception).isInstanceOf(AllDependenciesValidationException::class.java)
        val validationException = exception as AllDependenciesValidationException
        assertThat(validationException.status.description)
            .isEqualTo("Blocked service regular-1 from using all dependencies. Only defined services can use all dependencies")
        assertThat(validationException.status.code)
            .isEqualTo(Status.Code.INVALID_ARGUMENT)
```